internal/saucecloud/xctest.go

```diff
@@ -0,0 +1,228 @@
+	if passed {
+		exitCode = 0
+	}
+
+	return exitCode, nil
+}
+
+func (r *XctestRunner) dryRun() {
+	fmt.Println("\nThe following test suites would have run:")
+	for _, s := range r.Project.Suites {
+		fmt.Printf("  - %s\n", s.Name)
+		for _, d := range s.Devices {
+			fmt.Printf("    - on %s %s %s\n", d.Name, d.PlatformName, d.PlatformVersion)
+		}
+	}
+	fmt.Println()
+}
+
+func (r *XctestRunner) runSuites(ctx context.Context) bool {
+	jobOpts, results := r.createWorkerPool(ctx, r.Project.Sauce.Concurrency, r.Project.Sauce.Retries)
+	defer close(results)
+
+	suites := r.Project.Suites
+	if r.Project.Sauce.LaunchOrder != "" {
+		history, err := r.getHistory(ctx, r.Project.Sauce.LaunchOrder)
+		if err != nil {
+			log.Warn().Err(err).Msg(msg.RetrieveJobHistoryError)
+		} else {
+			suites = xctest.SortByHistory(suites, history)
+		}
+	}
+
+	// Submit suites to work on.
+	jobsCount := r.calculateJobsCount(suites)
+	go func() {
+		for _, s := range suites {
+			for _, d := range enumerateDevices(s.Devices, s.Simulators) {
+				log.Debug().Str("suite", s.Name).
+					Str("deviceName", d.name).Str("deviceID", d.ID).
+					Str("platformVersion", d.platformVersion).
+					Msg("Starting job")
+				r.startJob(jobOpts, s.App, s.XCTestRunFile, s.OtherApps, s, d)
+			}
+		}
+	}()
+
+	return r.collectResults(ctx, results, jobsCount)
+}
+
+func (r *XctestRunner) startJob(jobOpts chan<- job.StartOptions, appFileID, xcTestRunFileID string, otherAppsIDs []string, s xctest.Suite, d deviceConfig) {
+	jobOpts <- job.StartOptions{
+		ConfigFilePath:   r.Project.ConfigFilePath,
+		CLIFlags:         r.Project.CLIFlags,
+		DisplayName:      s.Name,
+		Timeout:          s.Timeout,
+		App:              appFileID,
+		XCTestRunFile:    xcTestRunFileID,
+		Suite:            xcTestRunFileID,
+		OtherApps:        otherAppsIDs,
+		Framework:        "xctest",
+		FrameworkVersion: "1.0.0-stable",
+		PlatformName:     d.platformName,
+		PlatformVersion:  d.platformVersion,
+		DeviceName:       d.name,
+		DeviceID:         d.ID,
+		Name:             s.Name,
+		Build:            r.Project.Sauce.Metadata.Build,
+		Tags:             r.Project.Sauce.Metadata.Tags,
+		Tunnel: job.TunnelOptions{
+			Name:  r.Project.Sauce.Tunnel.Name,
+			Owner: r.Project.Sauce.Tunnel.Owner,
+		},
+		Experiments:   r.Project.Sauce.Experiments,
+		TestsToRun:    s.TestOptions.Class,
+		TestsToSkip:   s.TestOptions.NotClass,
+		Attempt:       0,
+		Retries:       r.Project.Sauce.Retries,
+		PassThreshold: s.PassThreshold,
+		SmartRetry: job.SmartRetry{
+			FailedOnly: s.SmartRetry.IsRetryFailedOnly(),
+		},
+		TestOptions: s.TestOptions.ToMap(),
+
+		// RDC Specific flags
+		RealDevice:        d.isRealDevice,
+		DeviceHasCarrier:  d.hasCarrier,
+		DeviceType:        d.deviceType,
+		DevicePrivateOnly: d.privateOnly,
+
+		// VMD specific settings
+		Env:         s.Env,
+		ARMRequired: d.armRequired,
+
+		// Overwrite device settings
+		RealDeviceKind: strings.ToLower(xcuitest.IOS),
```

Comment:
Just for consistency (since this const exists in both packages):
```suggestion
		RealDeviceKind: strings.ToLower(xctest.IOS),
```